packages/amplify-graphql-transformer-core/src/transformer-context/resolver.ts

```diff
@@ -278,7 +278,10 @@ export class TransformerResolver implements TransformerResolverProvider {
           }
 
           if (context.isProjectUsingDataStore()) {
-            const syncConfig = SyncUtils.getSyncConfig(context, this.typeName)!;
+            //Remove the suffix "Table" from the datasource name
```

Comment:
One thing to illustrate is that the previous `this.typeName` is passed as `Query`, `Mutation`, `Subscription` instead of the model name, in which it will never get the model specific sync config.

Due to this, changing it to the model name is the right solution. However, I cannot find a good way to achieve this because
1. This model name is not passed into the class
2. The stack name is one possible way but it is tokenized and its value is not revealed by CDK synthesis process.

The fix is not final here. Please LMK if there are better approaches to pass in the model name values.

Reply:
I wonder if we can instead model the resolverConfig?.models as a CloudFormation mapping https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/mappings-section-structure.html and let CFN do the effective lookup.

Then the fact that you're dealing with a Token shouldn't matter I don't think.

Reply:
I paste the source code for creating the model table as datasource below. This string manipulation is subject to change once the following code is changed.
https://github.com/aws-amplify/amplify-category-api/blob/9a9b5e929de8126da2c6c058bbf93b79bf8db81c/packages/amplify-graphql-model-transformer/src/graphql-model-transformer.ts#L1272-L1290

Reply:
Like Al mentioned above, try to see if we can use `resolverConfig?.models` and if you think that's not working, I am okay with shipping it to unblock the customers.